common/djangoapps/edraak_social/forus/pipeline.py

```diff
@@ -0,0 +1,67 @@
+from social_core.pipeline.social_auth import associate_by_email
+from social_core.pipeline.user import create_user
+
+from student.models import UserProfile
+
+from .utils import forus_authentication_only
+
+
+@forus_authentication_only
+def save_redirect_param(strategy, *args, **kwargs):
+    strategy.request.session['next'] = \
+        strategy.request.GET.get('next', '').replace(' ', '+')
+
+
+@forus_authentication_only
+def associate_account_by_email(backend, details, user, *args, **kwargs):
+    """
+    This pipeline step associates the current social auth with the user which has the
+    same email address in the database.  It defers from social library's associate_by_email
+    implementation, which verifies that only a single database user is associated with the email.
+    """
+    return associate_by_email(backend, details, user, *args, **kwargs)
```

Comment:
I don't think we should allow `is_superuser` or `is_staff` to authenticate via Forus.